shared/src/service/transport.ts

```diff
@@ -101,13 +102,25 @@ export class ClientTransportService implements TransportService {
   private authenticatedPromise: Promise<void>;
   private authenticated = false;
 
-  constructor(private clientId: string) {}
+  /**
+   * If useAccessToken is set, you will need to call setAccessToken on the
+   * ClientTransportService instance before making your first request and
+   * call it again whenever the token expires
+   */
+
+  constructor(private clientId: string, private useAccessToken?: boolean) {}
 
   async submit<T>(request: ApiRequest): Promise<ApiResponse<T>> {
     await this.initializeClient();
     return gapi.client.request(request);
   }
 
+  async setAccessToken(accessToken: string): Promise<void> {
+    await this.initializeClient();
```

Comment:
See my previous comment. I wouldn't have the GAPI initialization here. I would just have it be a dumb setter (you could use the ES6 `set` style) for a private data member. The initialization flow would still occur at the time a request is made, where the GAPI library would be loaded. If the accessToken private data member was set, then it skips the next step.